### ISSUE TITLE
java.lang.StackOverflowError for over 2500 concurrent users

### ISSUE THREAD

**Issue by mlotysz** [mhjort/clj-gatling#40]
I'm trying to generate heavy load for some backend system but unfortunately I was hit by java.lang.StackOverflowError. Reproduction for this bug is super easy with your clj-gatling-example project.
```
$ lein run metrics 2800 1
Exception in thread "main" java.lang.StackOverflowError, compiling:(/private/var/folders/f5/k0cgb1rs0hldwslr_l5hlpkc0000gn/T/form-init8067742416466387465.clj:1:125)
	at clojure.lang.Compiler.load(Compiler.java:7391)
	at clojure.lang.Compiler.loadFile(Compiler.java:7317)
	at clojure.main$load_script.invokeStatic(main.clj:275)
	at clojure.main$init_opt.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invoke(main.clj:277)
	at clojure.main$initialize.invokeStatic(main.clj:308)
	at clojure.main$null_opt.invokeStatic(main.clj:342)
	at clojure.main$null_opt.invoke(main.clj:339)
	at clojure.main$main.invokeStatic(main.clj:421)
	at clojure.main$main.doInvoke(main.clj:384)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.StackOverflowError
	at clojure.core$seq__4357.invokeStatic(core.clj:137)
	at clojure.core$map$fn__4789.invoke(core.clj:2648)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:521)
	at clojure.core$seq__4357.invokeStatic(core.clj:137)
	at clojure.core$map$fn__4789.invoke(core.clj:2648)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:521)
	at clojure.core$seq__4357.invokeStatic(core.clj:137)
	at clojure.core$map$fn__4789.invoke(core.clj:2648)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:521)
...
```
Do you have any clues what is core problem here? I'd like to help.

**Comment by mlotysz** [comment]
Google reveals some hints: https://groups.google.com/forum/#!topic/clojure/ZoIwV8bzWfE

**Comment by mhjort** [comment]
Thanks for pointing this out. I can also reproduce this. I'll investigate a bit.

**Comment by mhjort** [comment]
I was able to fix this. https://github.com/mhjort/clj-gatling/pull/42

I will make patch release soon.

**Comment by mhjort** [comment]
Released version ```0.8.4``` that fixes the issue. 

Out of curiosity what kind of machine setup you are using for generating this load? I was wondering if you have considered using https://github.com/mhjort/clojider for this load?

**Comment by mlotysz** [comment]
The client machine is nothing special: AWS EC2 medium instance. I'm testing a simple scenario with one step only (OTP verification). I saw clojider and for sure I will use it next time. (AWS lambda should be easier to use than EC2.)

Thank you very much for your help and great project!
